torch/csrc/jit/serialization/unpickler.cpp

```diff
@@ -70,20 +70,12 @@ void restoreAccurateTypeTags(const IValue& root, const TypePtr& type_tag) {
       case LayoutType::Kind:
       case ScalarTypeType::Kind:
       case RRefType::Kind:
-        // no op, there is nothing to tag
-        break;
       case AnyType::Kind:
       case AnyListType::Kind:
       case AnyTupleType::Kind:
       case AnyClassType::Kind:
-        // if Any type does show up, we no longer have a way to precisely
-        // recover the type information since the w.value may be an untagged
-        // List/Dict. We should prevent objects being serialized from having the
-        // Any type and if we do allow it in functions limit it to non-heap
-        // locations.
-        TORCH_INTERNAL_ASSERT(
-            false,
-            "AnyType, AnyTupleType, AnyListType, and AnyClassType should not show up in the static type of objects");
+        // no op, there is nothing to tag
+        break;
```

Comment:
I think this change makes sense, but does it handle nested `Any` types, like `Dict[str, Any]` that has an actual, runtime value type of `Dict[str, Any]`?